src/main/java/org/simulator/math/QualityMeasure.java

```diff
@@ -159,73 +160,57 @@ public double distance(MultiTable x, MultiTable expected) {
     distances.addAll(getColumnDistances(left, right));
-    System.out.println("Current Distance Function results: " + meanFunction.computeMean(distances));
     return meanFunction.computeMean(distances);
   }
 
   /**
-   *
+   * Computes the distance of two matrices as the sum of the distances of each
+   * row. It is possible that one matrix contains more columns than the other
+   * one. If so, the additional values in the bigger matrix are ignored and do
+   * not contribute to the distance. {@link Double#NaN} values do also not
+   * contribute to the distance. Only columns with matching identifiers are
+   * considered for the distance computation.
+   * 
    * @param x
    * @param expected
-   * @return maxAbsDistances the list of maximum absolute distances in the blocks
+   * @return columnDistances the list of distances for the columns in the blocks
    */
-  public ArrayList<Double> getMaxAbsDistances(MultiTable x, MultiTable expected) {
+  public ArrayList<Double> getColumnDistances(MultiTable x, MultiTable expected) {
     ArrayList<Double> distances= new ArrayList<Double>();
     for(int block = 0; block < x.getBlockCount(); block++) {
       String identifiers[] = x.getBlock(block).getIdentifiers();
       for (int i = 0; i < identifiers.length; i++) {
         if ((identifiers[i]!=null) && (expected.getColumn(identifiers[i]) != null)) {
-          Column a = x.getBlock(block).getColumn(i);
-          Column b = expected.getColumn(identifiers[i]);
-          double x_i;
-          double y_i;
-          double d = Double.MIN_VALUE;
-
-          for (int z = 0 ; z < Math.min(a.getRowCount(), b.getRowCount()); z++){
-            x_i = a.getValue(z);
-            y_i = b.getValue(z);
-            if (!Double.isNaN(y_i) && !Double.isNaN(x_i) && (y_i != x_i)){
-              d = Math.max(d, Math.abs(x_i - y_i));
-            }
-          }
-
-          distances.add(d);
+          distances.add(distance(x.getBlock(block).getColumn(i), expected.getColumn(identifiers[i])));
         }
       }
     }
-
     return distances;
   }
 
   /**
-   * Computes the distance of two matrices as the sum of the distances of each
-   * row. It is possible that one matrix contains more columns than the other
-   * one. If so, the additional values in the bigger matrix are ignored and do
-   * not contribute to the distance. {@link Double#NaN} values do also not
-   * contribute to the distance. Only columns with matching identifiers are
-   * considered for the distance computation.
-   * 
+   *
    * @param x
    * @param expected
-   * @return columnDistances the list of distances for the columns in the blocks
+   * @return maxAbsDistances the list of maximum absolute distances in the blocks
    */
-  public ArrayList<Double> getColumnDistances(MultiTable x, MultiTable expected) {
-    ArrayList<Double> distances= new ArrayList<Double>();
+  public HashMap<String, Double> getMaxAbsDistances(MultiTable x, MultiTable expected) {
```

Comment:
Also here, please do not return `HashMap<String, Double>` but `Map<String, Double>` to make it easier to use a different implementation later on.

Reply:
@draeger, Updated this in the latest commit.